### DISCUSSION TITLE
Partial Records

### DISCUSSION THREAD

**Discussion by kine** [microsoft/alguidelines#74] [BC Patterns]
How to use correctly, when own local variable, when parameter from outside...

**Comment by waldo1001** [comment]
There is definitely something to say about this, absolutely.  What are you thinking of exactly, @kine ?

I'd refer to docs for the syntax and HOW to use it.
here, we can describe best practices .. .

I'm thinking of: Don't use when you intend to modify the record

but what else? 🤔

**Comment by kine** [comment]
How to use it e.g. when passing the rec as parameter (by ref) to another function (which I have under control or I do not have under control) and how to work if I am consuming record as parameter from outside (and I am in control of the call or I am not)

**Reply by kine** [reply]
It means, what is the best practice when the record variable is potentially crossing app boundaries (calling outside function, triggering public event, or is consumed in public function or in subscriber)

**Reply by waldo1001** [reply]
I get it - but isn't that more of a "how to", and less of a "best practice". 🤔
It IS very valid, though .. .

**Reply by kine** [reply]
That's what I struggle most - where is the line between "how to" and "best practice". May be it is better to be how to, best practice is habbit (mostly do it like that), "how to" is strict "do it in this way".

**Comment by waldo1001** [comment]
Correction - it seems that modifying a record is perfectly possible in combination with partial records.  It seems that this:

 ```AL
    procedure SetLoadFieldsAndModifyDifferentField()
    var
        JustSomeTableWLD: Record "Just Some Table WLD";
    begin
        SelectLatestVersion();

        JustSomeTableWLD.SetLoadFields(Message);
        JustSomeTableWLD.LockTable();
        JustSomeTableWLD.FindFirst();
        JustSomeTableWLD."Message 2" := Format(Random(1000));
        JustSomeTableWLD.Modify();
    end;
```
results into a simple partial SELECT and one UPDATE, and that's it.  It seems there is some tracking behind the scenes where the client knows which fields were modified .. ?

@KennieNP , @JesperSchulz , could you confirm?
